### PR TITLE
🐛  handle gitlab repos with no commits

### LINKED ISSUE #3727: BUG: Scanning a gitlab project with no commits crashes
**Describe the bug**
Probably an edge case, but attempting to scan a repo with no commits fails. 

```console
$ go run main.go --repo gitlab.com/*****/test-repo
Starting [Vulnerabilities]
Starting [Dependency-Update-Tool]
Starting [Binary-Artifacts]
Starting [Maintained]
Starting [Security-Policy]
Starting [Contributors]
Starting [CII-Best-Practices]
Starting [Packaging]
Starting [Branch-Protection]
Starting [Pinned-Dependencies]
Starting [License]
Starting [Token-Permissions]
Starting [Fuzzing]
Starting [Code-Review]
Starting [Signed-Releases]
Starting [CI-Tests]
Starting [Dangerous-Workflow]
Starting [SAST]
panic: runtime error: index out of range [0] with length 0
```

**Reproduction steps**
1. Create new GitLab repository (Do not initialize with a readme)
2. Set `GITLAB_AUTH_TOKEN` (if above repo is private)
3. run `scorecard --repo gitlab.com/myrepo`

**Expected behavior**
Scorecard runs/completes without error. Potentially returning a warning, but at least returning no score for anything commit based.

**Additional context**

### PR DESCRIPTION
**Pull request by ashearin**
#### What kind of change does this PR introduce?
Simple length check on raw commit array prior to attempting to access the first index

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

It's assumed that the repository will return commits, sets the first element as the 'before' time for future processing. 

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

Fixes #3727 
Related to #3193

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
Scorecard no longer crashes on GitLab repos with no commits
```

### CODE REVIEW

**Comment by spencerschrock** [comment]
Is draft status intentional?

**Comment by gabibguti** [comment]
Can we add a test for this `ListCommits` test case in `clients/gitlabrepo/client_test.go`?

File: clients/gitlabrepo/client.go [hunk 1]
```diff
@@ -180,11 +180,17 @@ func (client *Client) GetFileContent(filename string) ([]byte, error) {
...
 	if err != nil {
 		return []clients.Commit{}, err
 	}
 
-	before := commitsRaw[0].CommittedDate
+	if len(commitsRaw) < 1 {
+		return []clients.Commit{}, nil
+	} else {
+		before = commitsRaw[0].CommittedDate
+	}
```
Comment (spencerschrock) [inline clients/gitlabrepo/client.go:193]: In the interest of keeping the [happy path left](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88), can we get rid of the `else` block?

```go
if len(commitsRaw) < 1 {
	return []clients.Commit{}, nil
} 

before := commitsRaw[0].CommittedDate
```

File: clients/gitlabrepo/client_test.go [hunk 2]
```diff
@@ -69,3 +83,79 @@ func TestCheckRepoInaccessible(t *testing.T) {
...
+			name:         "No commits in repo",
+			responsePath: "./testdata/valid-checkruns-1", // Contains Empty Array
+			commits:      []clients.Commit{},
+			wantErr:      false,
+		},
+		{
+			name:         "Error in getMergeRequestDetail",
+			responsePath: "./testdata/valid-commits",
+			commits:      []clients.Commit{},
+			wantErr:      true,
```
Comment (spencerschrock) [inline clients/gitlabrepo/client_test.go:112]: do we know why getMergeRequestDetail is erroring, is it because the goodGatewayRoundTripper isn't returning a body?

I'm not sure this test is essential to the change, so up to you on refining it or cutting it.

**Review by spencerschrock — APPROVED** [review]
Change looks good, left two small comments about the tests.

File: clients/gitlabrepo/client_test.go [hunk 3]
```diff
@@ -69,3 +83,79 @@ func TestCheckRepoInaccessible(t *testing.T) {
...
+	}{
+		{
+			name:         "Error in ListRawCommits",
+			responsePath: "./testdata/invalid-commits",
+			commits:      []clients.Commit{},
+			wantErr:      true,
+		},
+		{
+			name:         "No commits in repo",
+			responsePath: "./testdata/valid-checkruns-1", // Contains Empty Array
```
[the remark this answers is not in the document]
Comment (ashearin) [reply clients/gitlabrepo/client_test.go:104]: I like it. Makes it much more reusable.

File: clients/gitlabrepo/client_test.go (hunk 2 above)
Comment (ashearin) [reply clients/gitlabrepo/client_test.go:112]: I think it's failing to validate the url info in the stub repourl we're passing in. Once we get around to making tests for gitlab graphql code we'll need to make a more robust test round tripper. 

I kept it in so I was covering both paths in the check for a non-zero commit array length. If you don't think it needed I'll remove it until we can tackle tests for `getMergeRequestDetail` and `zip`.
